Add new `consumerGroupRebalanceStrategy` and `useAvroJson` doc for Kafka pubsub (#4686)
dapr/docs: daprdocs/content/en/reference/components-reference/supported-pubsub/setup-apache-kafka.md

{{% alert title="Important" color="warning" %}}
Currently, only message value serialization/deserialization is supported. Since cloud events are not supported, the `rawPayload=true` metadata must be passed when publishing Avro messages.

Please note that `rawPayload=true` should NOT be set for consumers, as the message value will be wrapped into a CloudEvent and base64-encoded. Leaving `rawPayload` as default (i.e. `false`) will send the Avro-decoded message to the application as a JSON payload.

> When setting the `useAvroJson` component metatadata to `true`, the inbound/outbond Avro binary will be converted into/from Avro JSON encoding. 

Reviewer: ```suggestion
When setting the `useAvroJson` component metadata to `true`, the inbound/outbound Avro binary is converted into/from Avro JSON encoding. 
```
